PWGHF/TableProducer/HFTreeCreatorChicToJpsiGamma.cxx

```diff
@@ -0,0 +1,257 @@
+                  collision::BCId,
+                  full::Pt,
+                  full::Eta,
+                  full::Phi,
+                  full::Y,
+                  hf_cand_chic::JpsiToMuMuMass,
+                  full::MCflag);
+
+} // namespace o2::aod
+
+/// Writes the full information in an output TTree
+struct HfTreeCreatorChicToJpsiGamma {
+  Produces<o2::aod::HfCandChicFull> rowCandidateFull;
+  Produces<o2::aod::HfCandChicFullEvents> rowCandidateFullEvents;
+  Produces<o2::aod::HfCandChicFullParticles> rowCandidateFullParticles;
+
+  void init(InitContext const&)
+  {
+  }
+
+  void process(aod::Collisions const& collisions,
+               aod::McCollisions const& mccollisions,
+               soa::Join<aod::HfCandChic, aod::HfCandChicMCRec, aod::HFSelChicToJpsiGammaCandidate> const& candidates,
+               soa::Join<aod::McParticles, aod::HfCandChicMCGen> const& particles,
+               aod::BigTracksPID const& tracks,
+               aod::HfCandProng2 const& jpsiCands)
+  {
+
+    // Filling event properties
+    rowCandidateFullEvents.reserve(collisions.size());
+    for (auto& collision : collisions) {
+      rowCandidateFullEvents(
+        collision.bcId(),
+        collision.numContrib(),
+        collision.posX(),
+        collision.posY(),
+        collision.posZ(),
+        0,
+        1);
+    }
+
+    // Filling candidate properties
+    int indexCand = 0;
+    rowCandidateFull.reserve(candidates.size());
+    for (auto& candidate : candidates) {
+      // std::cout << "candidate: px, py, pz = " << candidate.px() << ", " << candidate.py() << ", " << candidate.pz() << std::endl;
+      // std::cout << "Prong 0: px, py, pz = " << candidate.pxProng0() << ", " << candidate.pyProng0() << ", " << candidate.pzProng0() << std::endl;
+      // std::cout << "Prong 1: px, py, pz = " << candidate.pxProng1() << ", " << candidate.pyProng1() << ", " << candidate.pzProng1() << std::endl;
+      // std::cout << "J/psi invariant mass: " << candidate.jpsiToMuMuMass() << std::endl;
+      // std::cout << "index 0, 1  = " << candidate.index0Id() << ", " << candidate.index1Id() << std::endl;
+      auto pxchic = candidate.px();
+      auto pychic = candidate.py();
+      auto pzchic = candidate.pz();
+      auto pchic = RecoDecay::P(candidate.px(), candidate.py(), candidate.pz());
+      auto pxjpsi = candidate.pxProng0();
+      auto pyjpsi = candidate.pyProng0();
+      auto pzjpsi = candidate.pzProng0();
+      auto pjpsi = RecoDecay::P(candidate.pxProng0(), candidate.pyProng0(), candidate.pzProng0());
+      auto pxgamma = candidate.pxProng1();
+      auto pygamma = candidate.pyProng1();
+      auto pzgamma = candidate.pzProng1();
+      auto pgamma = RecoDecay::P(candidate.pxProng1(), candidate.pyProng1(), candidate.pzProng1());
+      auto pl1 = abs(pxchic * pxjpsi + pychic * pyjpsi + pzchic * pzjpsi) / pchic;
+      auto pl2 = abs(pxchic * pxgamma + pychic * pygamma + pzchic * pzgamma) / pchic;
```

Comment:
Seems like this could be much simpler if you used `RecoDecay` functions for these calculations.

Reply:
done